consensus/core/src/core.rs

```diff
@@ -241,16 +239,27 @@ impl Core {
 
     /// Attempts to propose a new block for the next round. If a block has already proposed for latest
     /// or earlier round, then no block is created and None is returned.
-    fn try_new_block(&mut self, ignore_leaders_check: bool) -> Option<VerifiedBlock> {
+    fn try_new_block(&mut self, force: bool) -> Option<VerifiedBlock> {
         let _scope = monitored_scope("Core::try_new_block");
+
         let clock_round = self.threshold_clock.get_round();
         if clock_round <= self.last_proposed_round() {
             return None;
         }
+
+        let now = timestamp_utc_ms();
+
         // Create a new block either because we want to "forcefully" propose a block due to a leader timeout,
-        // or because we are actually ready to produce the block (leader exists)
-        if !(ignore_leaders_check || self.last_quorum_leaders_exist()) {
-            return None;
+        // or because we are actually ready to produce the block (leader exists and min delay has passed).
+        if !force {
+            if !self.last_quorum_leaders_exist() {
+                return None;
+            }
+            if Duration::from_micros(now.saturating_sub(self.last_proposed_timestamp_ms()))
+                < self.context.parameters.min_round_delay
+            {
```

Comment:
My understanding here is that if all the blocks of a round are received in <50ms from last proposal then we eventually rely on the leader timeout logic to move the round. I am just curious if even for our tests this would be the common case . That also gives a bit of additional semantics on the leader timeout component as it now also guarantees liveness under the short round cases too.

The other point we should pay attention to is to make sure that the leader timeout component respects a min round delay as well. This might require a small refactoring on my PR https://github.com/MystenLabs/sui/pull/16606 to ensure this

Reply:
The currently logic ignores min round delay when "forcing" a new block. I modified `test_core_try_new_block_leader_timeout` a bit to verify this scenario.

Reply:
Yes agreed on that and that's why we won't lose liveness. My main point was on the fact that the leader timeout component is the one that guarantees liveness now even in case where we have all the leaders, but we received all the round blocks too fast.

Reply:
Yeah currently leader timeout drives proposing if the previous round leader is missing. This change starts to require leader timeout to drive proposing when leader arrives before min round delay as well.